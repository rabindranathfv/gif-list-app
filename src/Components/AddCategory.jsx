import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = (props) => {

    const { setCategory } = props;
    const [inputValue, setInputValue] = useState('');

    const addHandler = (e) => {
        setInputValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if ( inputValue.trim().length > 2) {
            setCategory(cat => [ inputValue]);
            console.log('AQUI');
            setInputValue('');
            console.log('AQUI 4');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            { inputValue && <p>{ inputValue }</p> }
            <input type="text" value={inputValue} onChange={addHandler} />
        </form>
    )
}

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}