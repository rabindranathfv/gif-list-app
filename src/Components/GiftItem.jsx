import React from 'react'

export const GiftItem = ({ gift }) => {

    return (
        <div key={gift.id} className='card'>
            <p > {gift.title}</p>  
            <img src={gift.url} alt={gift.title} />
        </div>
    )
}
