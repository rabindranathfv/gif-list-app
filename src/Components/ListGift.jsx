import React, { useState } from 'react';

export const ListGift = (props) => {

    const { gifts } = props;
    const [categories, setCategories] = useState(gifts);
    return (
        <div>
            <ul>
            {
                categories.map( cat => {
                    return <li key={cat.id}> {cat.name} </li>
                })
            }
            </ul>
        </div>
    )
}
