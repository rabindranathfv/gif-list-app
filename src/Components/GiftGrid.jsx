import React, { useEffect, useState } from 'react'
import PropTypes, { string } from 'prop-types';

import { GiftItem } from './GiftItem';

const API_URL = `https://api.giphy.com/v1/gifs/search`;
const API_KEY = `6ijpJmhvX9OshRF2lAHuBYHojzB9bZjX`;

export const GiftGrid = props => {

    const { category, searchCategory } = props;

    const [gifts, setGifts] = useState([]);

    const getGifts = async() => {
        const urlQuery = `${API_URL}?q=${searchCategory}&limit=10&api_key=${API_KEY}`
        const resp = await fetch( urlQuery);
        const { data } = await resp.json();
        
        const giftItem = data.map( gif => {
            return { 
                id: gif.id,
                title: gif.title,
                url:  gif.images?.downsized_medium.url
            }
        })
        setGifts( giftItem);
    }

    useEffect(() => {
        getGifts();
    }, []);

    return (
        <div className='card-grid'>
            <ul>
            {
                gifts.map( (gif) => {
                    return <GiftItem gift={gif} /> 
                })
            }
            </ul>
        </div>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.shape({
        id: string,
        title: string,
        url: string
      }).isRequired,
}
