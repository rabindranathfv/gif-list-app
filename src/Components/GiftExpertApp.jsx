import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GiftGrid } from './GiftGrid';

const GiftExpertApp = () => {

    const [category, setCategory] = useState('');

    console.log('CATEGORY IN MAIN COMPONENT', category);

    return (
        <div>
            <h2> GifExpertAPP </h2>
            <AddCategory setCategory={setCategory} />
            <hr />
            <GiftGrid category={category}  />
        </div>
    )
}

GiftExpertApp.defaultProps = {
    category: 'dragon ball',
}

export default GiftExpertApp
