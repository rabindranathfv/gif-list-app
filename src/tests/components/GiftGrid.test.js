import { shallow } from 'enzyme';
import React from 'react';
import { GiftGrid } from '../../Components/GiftGrid';

jest.mock('../../hooks/useFetchGift',
  () => {
    return {
      useFetchGift: () => {
        return {
          gifts: useFetchGiftMock,
          loading: false
        }
      }
    }
  }
);

const useFetchGiftMock = [
    {
      id: 'GRSnxyhJnPsaQy9YLn',
      title: 'Dragon Ball GIF by TOEI Animation UK',
      url: 'https://media4.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=59b5fe6594zonf5h4rx9rvzt7vuiaytclcniik77pzubyfu5&rid=giphy.gif&ct=g'
    },
    {
      id: '977YesTjNfQC7vQiph',
      title: 'Dragon Ball GIF by TOEI Animation UK',
      url: 'https://media2.giphy.com/media/977YesTjNfQC7vQiph/giphy.gif?cid=59b5fe6594zonf5h4rx9rvzt7vuiaytclcniik77pzubyfu5&rid=giphy.gif&ct=g'
    }
];

describe('should test GiftGrid component', () => {
    const category = ['dragon ball'];
    const wrapper = shallow( <GiftGrid category={category} />);
    
    it('should render giftGrid component' , () => {
      expect(wrapper).toMatchSnapshot();
   });

   it('should giftGrid component have a category in a title' , () => {
        const categoryExist = wrapper.find('h3');
        expect( typeof categoryExist.text().trim() ).toEqual('string');
        expect( categoryExist.text().trim()).toEqual('dragon ball');
    });

    it('should giftGrid component have a class without response' , () => {
        const classNameGifts = wrapper.find('.grid-gift');
        expect(classNameGifts).toBeDefined();
        expect(classNameGifts).toHaveLength(1);
        expect(classNameGifts.props().className).toEqual('grid-gift');
    })

    it('should giftGrid component have a class with success custom hook response' , () => {
      const wrapper = shallow( <GiftGrid category={category} />);
      expect(wrapper).toMatchSnapshot();
      
      const GiftItem = wrapper.find('GiftItem');
      GiftItem.debug();
      expect(GiftItem.length).toBe(useFetchGiftMock.length);
  })
});
