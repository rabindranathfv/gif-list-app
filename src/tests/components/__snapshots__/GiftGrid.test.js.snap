// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test GiftGrid component should render giftGrid component 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
     
    dragon ball
     
  </h3>
  <p>
    loading...
  </p>
  <div
    className="grid-gift"
  />
</Fragment>
`;
